lib/internal/event_target.js

```diff
@@ -334,14 +334,13 @@ class EventTarget {
       }
       return event;
     };
+    if (event !== undefined)
```

Comment:
Good catch! If you wouldn't mind adding a test for this (dispatching an event that has no listeners should still set the `target`) it would be appreciated. If not I'll push one later.

Reply:
We have a test case in WPT already as it was found by the below case :)
https://github.com/nodejs/node/blob/29cf5a15171d3187622b31426d7e3c844ea0aa1e/test/fixtures/wpt/dom/abort/event.any.js#L60

In `whatwg-url` or some of the other modules using WPT tests, we usually try to add new test to upstream WPT repository when we find any missing cases and I think the same strategy can be applied to `AbortController` once this PR is merged.

Reply:
I think it makes sense to also test this for EventTarget and not just for AbortController - though I haven't been able to port the WPT Events tests when I tried because they were used `Node` rather than `EventTarget`. I added a few manually but not a lot.

Reply:
Sounds good. I added a case to `test-eventtarget.js` :)